### ISSUE TITLE
[Accessibility] Tab and tabpanel needs appropriate name and role information

### ISSUE THREAD

**Issue by Krituhar** [microsoft/vscode#227496]
- VS Code Version: 1.88.0
- OS Version: MacOS Sonoma (v14.6.1)

**[Issue]**
The "Hello.txt" tab and its tabpanel is without appropriate name and role information. The  aria-controls for the tab and the tabpanel structure is missing.

**[User Impact]**
Screen reader users will be unable to determine that these controls reveal panels of content and which panel is currently revealed.

**[Code Reference]**
```
<div role="tablist" draggable="true" class="tabs-container" style="overflow: hidden;"><div draggable="true" role="tab" class="tab tab-actions-right sizing-fit has-icon tab-border-top active" aria-label="index.mjs, preview" aria-description="" data-resource-name="index.mjs" aria-selected="true" tabindex="0" ...>
...
</div></div>
<div role="tablist" draggable="true" class="tabs-container" style="overflow: hidden;">...</div>
<div class="editor-container" style="background-color: rgb(255, 255, 255); height: 486px;">...</div>
```

**[Recommendation]**
Ensure page tabs provide state and role.

For tabs, the following information is expected:
- Each panel container must have role="tabpanel".
- If the tablist has a visible label, the tablist element must have aria-labelledby set to the ID of the labelling element. Otherwise, the tablist element must have aria-label set to the accessible name.
- Each tab must have aria-controls set to the ID of its corresponding tabpanel.
- Tabpanel elements must have aria-labelledby set to the ID of their corresponding tab.

Screenshot of the issue:

![Screenshot 2024-09-03 at 5 29 37 PM](https://github.com/user-attachments/assets/89135e94-eb12-428f-9ddb-49707dc96a52)

**Comment by meganrogge** [comment]
No users have reported this - thus closing
